pkg/tenant/mock/quarterdeck_test.go

```diff
@@ -0,0 +1,80 @@
+package mock_test
+
+import (
+	"encoding/json"
+	"net/http"
+	"testing"
+
+	"github.com/rotationalio/ensign/pkg/quarterdeck/api/v1"
+	"github.com/rotationalio/ensign/pkg/tenant/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMock(t *testing.T) {
+	// Start the mock server
+	quarterdeck, err := mock.NewServer()
+	require.NoError(t, err, "could not start the mock quarterdeck server")
+	t.Cleanup(func() {
+		quarterdeck.Close()
+	})
+	client := quarterdeck.Client()
+
+	// Unconfigured path should return 404
+	req, err := http.NewRequest(http.MethodGet, quarterdeck.URL()+"/v1/invalid", nil)
+	require.NoError(t, err, "could not create request")
+	rep, err := client.Do(req)
+	require.NoError(t, err, "could not execute request")
+	require.Equal(t, http.StatusNotFound, rep.StatusCode, "expected status code to be 404")
+
+	// Try a request with a default handler
+	quarterdeck.OnStatus()
+	req, err = http.NewRequest(http.MethodGet, quarterdeck.URL()+"/v1/status", nil)
+	require.NoError(t, err, "could not create request")
+	rep, err = client.Do(req)
+	require.NoError(t, err, "could not execute request")
+	require.Equal(t, http.StatusOK, rep.StatusCode, "expected status code to be 200")
+
+	// Try request with a different status code
+	quarterdeck.OnStatus(mock.UseStatus(http.StatusServiceUnavailable))
```

Comment:
Hopefully this will make the test code readable and easier to configure mocked responses from Quarterdeck.

Reply:
This makes a lot of sense!